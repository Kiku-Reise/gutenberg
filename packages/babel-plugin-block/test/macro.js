/**
 * External dependencies
 */
import plugin from 'babel-plugin-macros';
import pluginTester from 'babel-plugin-tester';

pluginTester( {
	plugin,
	babelOptions: {
		babelrc: false,
		filename: __filename,
		presets: [ '@wordpress/babel-preset-default' ],
	},
	tests: {
		'valid metadata file': {
			code: `
				import getBlockData from '../macro';
				const metadata = getBlockData( './fixtures/block.json' );
			`,
			snapshot: true,
		},
		'valid metadata file with i18n support': {
			code: `
				import getBlockData from '../macro';
				const metadata = getBlockData( './fixtures/block-i18n.json' );
			`,
			snapshot: true,
		},
		'valid metadata file with i18n support and default text domain': {
			code: `
				import getBlockData from '../macro';
				const metadata = getBlockData( './fixtures/block-i18n-default.json' );
			`,
			snapshot: true,
		},
		'invalid metadata file name': {
			code: `
				import getBlockData from '../macro';
				const metadata = getBlockData( './invalid-file.json' );
			`,
			error: 'Invalid file name provided: packages/babel-plugin-block/test/invalid-file.json.',
		},
		'invalid usage: as function argument': {
			code: `
				import getBlockData from '../macro';
        		const metadata = doSomething( getBlockData );
      		`,
			error: true,
		},
		'invalid usage: missing file path': {
			code: `
				import getBlockData from '../macro';
				const metadata = getBlockData;
			`,
			error: true,
		},
	},
} );
