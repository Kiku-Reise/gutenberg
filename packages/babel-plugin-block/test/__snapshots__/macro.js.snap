// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros valid metadata file with i18n support and default text domain: valid metadata file with i18n support and default text domain 1`] = `
"
import getBlockData from '../macro';
const metadata = getBlockData( './fixtures/block-i18n-default.json' );

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _i18n = require(\\"@wordpress/i18n\\");

const metadata = {
  name: \\"my-plugin/notice\\",
  title: (0, _i18n._x)(\\"Notice\\", \\"block title\\", \\"my-plugin\\"),
  category: \\"common\\",
  description: (0, _i18n._x)(\\"Shows warning, error or success notices  ...\\", \\"block description\\", \\"my-plugin\\"),
  keywords: [(0, _i18n._x)(\\"alert\\", \\"block keywords\\", \\"my-plugin\\"), (0, _i18n._x)(\\"message\\", \\"block keywords\\", \\"my-plugin\\")],
  attributes: {},
  styles: [{
    name: \\"default\\",
    label: (0, _i18n._x)(\\"Default\\", \\"block styles\\", \\"my-plugin\\"),
    isDefault: true
  }, {
    name: \\"other\\",
    label: (0, _i18n._x)(\\"Other\\", \\"block styles\\", \\"my-plugin\\")
  }]
};
"
`;

exports[`macros valid metadata file with i18n support: valid metadata file with i18n support 1`] = `
"
import getBlockData from '../macro';
const metadata = getBlockData( './fixtures/block-i18n.json' );

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _i18n = require(\\"@wordpress/i18n\\");

const metadata = {
  name: \\"my-plugin/notice\\",
  title: (0, _i18n._x)(\\"Notice\\", \\"block title\\"),
  category: \\"common\\",
  description: (0, _i18n._x)(\\"Shows warning, error or success notices  ...\\", \\"block description\\"),
  keywords: [(0, _i18n._x)(\\"alert\\", \\"block keywords\\"), (0, _i18n._x)(\\"message\\", \\"block keywords\\")],
  attributes: {},
  styles: [{
    name: \\"default\\",
    label: (0, _i18n._x)(\\"Default\\", \\"block styles\\"),
    isDefault: true
  }, {
    name: \\"other\\",
    label: (0, _i18n._x)(\\"Other\\", \\"block styles\\")
  }]
};
"
`;

exports[`macros valid metadata file: valid metadata file 1`] = `
"
import getBlockData from '../macro';
const metadata = getBlockData( './fixtures/block.json' );

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

const metadata = {
  name: \\"my-plugin/notice\\",
  title: \\"Notice\\",
  category: \\"common\\",
  parent: [\\"core/group\\"],
  icon: \\"star\\",
  description: \\"Shows warning, error or success notices  ...\\",
  keywords: [\\"alert\\", \\"message\\"],
  attributes: {
    message: {
      type: \\"string\\",
      source: \\"html\\",
      selector: \\".meessage\\"
    }
  },
  styles: [{
    name: \\"default\\",
    label: \\"Default\\",
    isDefault: true
  }, {
    name: \\"other\\",
    label: \\"Other\\"
  }]
};
"
`;
